src/python/pants/backend/python/typecheck/pyright/rules.py

```diff
@@ -71,6 +80,58 @@ class PyrightPartitions(Collection[PyrightPartition]):
     pass
 
 
+@rule_helper
+async def _patch_config_file(
+    config_files: ConfigFiles, venv_dir: str, source_roots: Iterable[str]
+) -> Digest:
+    """Patch the Pyright config file to use the incoming venv directory (from
+    requirements_venv_pex). If there is no config file, create a dummy pyrightconfig.json with the
+    `venv` key populated.
+
+    The incoming venv directory works alongside the `--venv-path` CLI argument.
+
+    Additionally, add source roots to the `extraPaths` key in the config file.
+    """
```

Comment:
Consider warning or erroring if these values are already set in the config, so that folks don't wonder why they are being overwritten/ignored.

Reply:
This feeds back into here: https://github.com/pantsbuild/pants/discussions/17748

I don't really know the best approach here - currently `venv` is being overwritten (destructive), otherwise it can't work. `extraPaths` is being added to (non-destructive). 

Throwing a warning is fine, but errors reduce from legitimate usability for anyone using/doing anything outside of Pants as well while dev'ing. e.g. I might run `pytest` out-of-band of Pants as well, so I'd want warnings to be suppressed.

Reply:
@thejcannon Any thoughts on this?